test/e2e/steps/core_steps.js

```diff
@@ -71,71 +72,102 @@ module.exports = function coreSteps () {
     })(this))
   })
 
-  this.When(/^I (run|runOut|start|init|stop) Karma( with log-level ([a-z]+))?$/, function (command, withLogLevel, level, callback) {
-    this.writeConfigFile(tmpDir, tmpConfigFile, (function (_this) {
-      return function (err, hash) {
-        if (err) {
-          return callback.fail(new Error(err))
-        }
-        level = withLogLevel === undefined ? 'warn' : level
-        var configFile = path.join(tmpDir, hash + '.' + tmpConfigFile)
-        var runtimePath = path.join(baseDir, 'bin', 'karma')
-        var execKarma = function (done) {
-          var cmd = runtimePath + ' ' + command + ' --log-level ' + level + ' ' + configFile + ' ' + additionalArgs
-
-          return exec(cmd, {
-            cwd: baseDir
-          }, done)
-        }
-        var runOut = command === 'runOut'
-        if (command === 'run' || command === 'runOut') {
-          _this.child = spawn('' + runtimePath, ['start', '--log-level', 'warn', configFile])
-          var done = function () {
-            cleansingNeeded = true
-            _this.child && _this.child.kill()
-            callback()
+  this.When(/^I (run|runOut|start|init|stop) Karma( with log-level ([a-z]+))?( behind a proxy on port ([0-9]*) that prepends '([^']*)' to the base path)?$/, function (command, withLogLevel, level, behindProxy, proxyPort, proxyPath, callback) {
```

Comment:
Ah I misread - I was worrying about the size of the step you're worrying about the size of the file. That's easier to deal with :)